web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileBlock.php

```diff
@@ -75,14 +76,10 @@ public function build() {
     // exceptions and log them so the site can continue running and display an
     // empty table instead of a fatal error.
     $profile = NULL;
-    $username = $this->getContext('username')->getContextData()->getValue();
     try {
-      $filter = [
-        'where' => [
-          'username' => $username,
-        ],
-      ];
-      $profile = $this->profileApi->profileFindOne(json_encode($filter));
+      $profile = $this->profileRepository->getProfileByUsername(
+        $this->getContextValue('username')
+      );
```

Comment:
To be honest I think PhpStorm did it for me automatically when editing the line.

I am not sure PHPCS would actually complain about it.